bdai_ros2_wrappers/bdai_ros2_wrappers/action.py

```diff
@@ -0,0 +1,583 @@
+
+        Raises:
+            RuntimeError: if action feedback tracking is not enabled,
+            action acknowledgement has not been received yet, or
+            action was not accepted.
+        """
+        if self._feedback_tape is None:
+            raise RuntimeError("Action feedback tracking is not enabled")
+        if not self._goal_handle_future.done():
+            raise RuntimeError("Action not acknowledged yet")
+        goal_handle = self._goal_handle_future.result()
+        if not goal_handle.accepted:
+            raise RuntimeError("Action not accepted")
+        outerframe = inspect.stack()[1]
+        yield from self._feedback_tape.content(
+            follow=True, timeout_sec=timeout_sec, label=f"{outerframe.filename}:{outerframe.lineno}"
+        )
+
+    @property
+    def accepted(self) -> bool:
+        """Check if action was accepted."""
+        return not self._goal_handle_future.done() or self._goal_handle_future.result().accepted
+
+    @property
+    def cancelled(self) -> bool:
+        """Check if action was cancelled."""
+        return (
+            self._result_future is not None
+            and self._result_future.done()
+            and (self._result_future.result().status == action_msgs.msg.GoalStatus.STATUS_CANCELED)
+        )
+
+    @property
+    def aborted(self) -> bool:
+        """Check if action was aborted."""
+        return (
+            self._result_future is not None
+            and self._result_future.done()
+            and (self._result_future.result().status == action_msgs.msg.GoalStatus.STATUS_ABORTED)
+        )
+
+    @property
+    def succeeded(self) -> bool:
+        """Check if action was succeeded."""
+        return (
+            self._result_future is not None
+            and self._result_future.done()
+            and (self._result_future.result().status == action_msgs.msg.GoalStatus.STATUS_SUCCEEDED)
+        )
+
+    @property
+    def status(self) -> Optional[int]:
+        """
+        Get action status code.
+
+        See `action_msgs.msg.GoalStatus` documentation for status codes.
+        Rejected actions are conventially assigned `None` for status.
+
+        Raises:
+            RuntimeError: if action has not been acknowledged or is still executing.
+        """
+        if not self._goal_handle_future.done():
+            raise RuntimeError("Action not acknowledged yet")
+        if not self._goal_handle_future.result().accepted:
+            return None
+        if self._result_future is None or not self._result_future.done():
+            raise RuntimeError("Action still executing")
+        return self._result_future.result().status
+
+    def as_future(self) -> Future:
+        """
+        Get action future as a plain future.
+
+        Useful to pass action futures to APIs that expect plain futures.
+        """
+        future = Future(executor=self._executor_ref())
+
+        def _done_callback(_: Future) -> None:
```

Comment:
nit: for clarity
```suggestion
        def _done_callback(_: Future) -> None:
            nonlocal future
```

Reply:
Done in https://github.com/bdaiinstitute/ros_utilities/pull/58/commits/329efc3935f1f9fdc187d6157e4d218f4289adfd.